contrib/olivere/elastic/option.go

```diff
@@ -46,3 +48,10 @@ func WithAnalyticsRate(rate float64) ClientOption {
 		cfg.analyticsRate = rate
 	}
 }
+
+// WithResourceNamer sets a function to set the span's resource name
+func WithResourceNamer(namer func(url, method string) string) ClientOption {
```

Comment:
I'm not so fond of the name you've used, but not strongly against it. In our system, it's pretty common to use the word "quantizer" in this scenario. Can we call it `WithQuantizer`? Also, we should document this a bit better, e.g.:
```go
// WithQuantizer specifies a quantizing function which will be used to obtain a resource name for a given
// ElasticSearch request, using the request's URL and method. Note that the default quantizer obfuscates
// IDs and indexes and by replacing it, sensitive data could possibly be exposed, unless the new quantizer
// specifically takes care of that.
func WithQuantizer(q func(url, method string) string) ClientOption {
```
If you think quantizer is confusing, I'm open to reconsidering this suggestion.

Reply:
Without having read the code, I'd be confused by a `WithQuantizer` option, but the more detailed documentation would help. Want me to change it?

Thanks for the doc suggestions, too.

Reply:
No, you can leave yours - the more human-friendly one 😄. About the docs, yes, please do.

Reply:
:) Fixed the docs